### DISCUSSION TITLE
Airflow on ARM - Optional provider feature disabled when importing 'airflow.providers.google.leveldb.hooks.leveldb.LevelDBHook' from 'apache-airflow-providers-google' package

### DISCUSSION THREAD

**Discussion by kijewskimateusz** [apache/airflow#24383] [General]
### Apache Airflow version

2.3.2 (latest released)

### What happened

I'm running locally Airflow 2.3.2 on my ARM-based laptop and when building webserver, I'm seeing multiple notifications with content:

`{providers_manager.py:218} INFO - Optional provider feature disabled when importing 'airflow.providers.google.leveldb.hooks.leveldb.LevelDBHook' from 'apache-airflow-providers-google' package`

### What you think should happen instead

I think info should be prompted only once or feature should be enabled

### How to reproduce

I'm using `apache/airflow:2.3.2` image, installing additional provider packages mentioned below

### Operating System

12.4 (21F79)

### Versions of Apache Airflow Providers

apache-airflow-providers-postgres==4.1.0
apache-airflow-providers-databricks==2.6.0

### Deployment

Docker-Compose

### Deployment details

I'm building custom dockerimage as provided in Airflow docupmmentation and then execution it using docker compose command

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by kijewskimateusz** [comment]
I've tested it with `airflow:slim-2.3.2` image and issue disappeared. After adding `apache-airflow-providers-google==7.0.0
` into Airflow dependencies, I'm notified about optional provider feature being disabled.

I'm wondering whether this won't cause any harm when needed to use this module using ARM-based machine

**Comment by potiuk** [comment]
> I've tested it with `airflow:slim-2.3.2` image and issue disappeared. After adding `apache-airflow-providers-google==7.0.0 ` into Airflow dependencies, I'm notified about optional provider feature being disabled.
> 
> I'm wondering whether this won't cause any harm when needed to use this module using ARM-based machine

INFO is an info - not warning. It's for diagnostics.  I don't think INFO message is anything you should be worried about.

You can get rid of it by installing the optional dependencies. But maybe indeed some extra message how it can be enabled. Would you like to maybe contribute a better message there? 

I am happy to guide you and review your Pull Request for it. You can become one of > 2K contributors to Apache Airflow.

Converting it to a discussin as this is really not a bug.

**Comment by ryanburda** [comment]
Ran into the same thing with `airflow.providers.google.cloud.hooks.cloud_sql.CloudSQLHook`.

Following @potiuk 's advice I was able to resolve the issue by manually importing that module to find out what dependencies were missing.
```
from airflow.providers.google.cloud.hooks.cloud_sql import CloudSQLHook
```
[apache-airflow-providers-mysql](https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/hooks/cloud_sql.py#L52) ended up being the issue. The INFO message no longer showed up after adding that dependency.

**Reply by aprettyloner** [reply]
Also running into this on 2.4.3.

To reproduce, add a no-op initContainer to scheduler with `image: apache/airflow:2.4.3`.

```
Optional provider feature disabled when importing 'airflow.providers.google.leveldb.hooks.leveldb.LevelDBHook' from 'apache-airflow-providers-google' package
```

**Reply by potiuk** [reply]
So ?

**Reply by potiuk** [reply]
Is info about optional feature disabled a problem ? Why?

**Reply by AaronCoquet-Easypark** [reply]
Info about `disabled optional features` is not a problem. Having it show up multiple times in one call is annoying, and not knowing how to fix it is frustrating.

**Reply by potiuk** [reply]
Fair enough. I will convert it back into issue and mark it as good first issue. If you would like to fix /improve it @AaronCoquet-Easypark - you are most welcome to.
